src/lib/features/metrics/instance/instance-service.ts

```diff
@@ -99,24 +98,33 @@ export default class ClientInstanceService {
         );
     }
 
+    private updateSeenClient = (data: IClientApp) => {
+        const current = this.seenClients[this.clientKey(data)];
+        this.seenClients[this.clientKey(data)] = {
+            ...current,
+            ...data,
+        };
+    };
+
     public async registerInstance(
-        data: PartialSome<IClientApp, 'instanceId'>,
+        data: Pick<
+            ClientMetricsSchema,
+            'appName' | 'instanceId' | 'environment'
+        >,
         clientIp: string,
     ): Promise<void> {
-        const value = await clientMetricsSchema.validateAsync(data);
```

Comment:
is this now done by openapi?

Reply:
Partially. 

- OpenAPI guarantees that `instanceId` and `appName` are there. These are the bits we care the most about. 
- There's also `environment` which is optional in OpenAPI, but which clientMetricsSchema.validate required. This might be useful if it's not already present, but it'll show up when you register the client anyway. Besides, I don't know when it wouldn't be there. Finally, even if it's not there, we can manually default it to `default`, which is what the validation method did. (Although I'm leaning towards removing it in a follow-up)
- The real reason I removed it is that that schema validation also requires your input to have `buckets: {...}` defined, which a) we don't use, and b) is a pain to set up during testing.

So yes. OpenAPI takes care of what we need it to.